src/utils/mavenUtils.ts

```diff
@@ -16,6 +20,16 @@ export namespace mavenUtils {
         try {
             await cpUtils.executeCommand(undefined, workingDirectory, mvnCommand, '--version');
         } catch (error) {
+            const settingKey: string = 'showMavenPathWarning';
+            if (getFuncExtensionSetting<boolean>(settingKey)) {
+                const message: string = localize('azFunc.mvnNotFound', 'Failed to find "maven", please ensure that the maven bin directory is in your system path.');
```

Comment:
I don't think a "Don't warn again" button is appropriate in this case:
1. This message is very similar to the error you throw on line 33, but better. I would rather have this show every time than that error
1. This is really an error, not a warning. We don't hide this kind of error from the user (aka an error that blocks them from continuing)

Reply:
That's a valid concern. Also I've noticed that we will have two toasts here:
- One is for trouble shotting
- Another is the original one `Failed to find "maven", please ensure ...`

@EricJizbaMSFT Is it possible that we can only have one toast here?

Reply:
Well one option is not to throw an error. But obviously then you don't get it in telemetry, etc. 

That's why we have the `actionContext.suppressErrorDisplay` property. Here's an example: https://github.com/Microsoft/vscode-azurefunctions/blob/b56a2b0ccddac99993891204250fc17f6898ad16/src/commands/createNewProject/validateFunctionProjects.ts#L33

Reply:
@EricJizbaMSFT Thanks for your advice, and please have a review.